### DISCUSSION TITLE
Reaching out and engagement

### DISCUSSION THREAD

**Discussion by longr** [NERC-CEH/rse_group#27] [General]
I wanted somewhere to collate a list of ideas for how we can start engaging with the wider organisation:

**Code clinic** - Block of time with a regular scheduled meeting.  I have done in person before, people turn up with a issue / problem and you triage and help (<15 mins per problem, otherwise becomes a small project and goes that route).   Other RSE groups have done this remote.  1 Facilitator so there is someone in the main room, and then go to breakout rooms with people to discuss issues.

**RSE Lunch**  Similar to code clinic, but more chatting and less typing.  Perhaps best in person at each site?  Offer times when you are willing to have a working group (2 RSEs) and let people know where and when so they can join and chat about RSE stuff.

**Research Software Forums**, this is the name we used where I was at before.   5 minute talks on methods/tools/techniques (often given by RSEs or researchers with expertise in those areas, and 5 minute Science talks focussing on how they use software or workflows - either to share or ask advice.  Time for discussion afterwards. 

I think these could be done monthly or quarterly (perhaps not all of them).    RSE Lunch would be an issue, but would be happy to do remotes for Bangor.

**Comment by longr** [comment]
Any other suggestions or thoughts on these?

**Comment by metazool** [comment]
A hearty +1 for the "code clinic" as a low-pressure way of building community amongst people whose main job it is to write code. 

[Notes on how this is running at BGS](https://github.com/NERC-CEH/rse_group/blob/main/learning/clinics.md) after a recent visit to observe. There's a wide constituency here including data stewards and core developers as well as research coders.

It probably works best if it's prefaced by a short livecode-style interactive tutorial - the one I recently attended covered all the different styles of string formatting in python, or could be intro to code linting, creating a minimal package, working with an interactive debugger. It's a forum for discovering problems, but err on the side of helping people to help themselves. _Ideally_ an RSE group would coordinate rather than always deliver it :D 

I've heard that in-person is a much happier experience than remote for this.

**Comment by metazool** [comment]
https://aberdeenstudygroup.github.io/studyGroup/#whatwedo - @albags mentioned this in the weekly wider catchup, a "study group" which current Wallingford staff were involved in running; there is already an ongoing discussion at that site about setting up a study group with initial aims of supporting research coders who do primarily R to pick up python

**Comment by mattjbr123** [comment]
@fsamreen's summary of the NC8CIR-DRI event they attended ~Apr2024, containing some ideas in line with the above:

> Several panel discussions focused on the RSE service model and outreach activities, highlighting the key role RSEs play in sharing best practices, supporting proof of concepts, and raising awareness about available technologies and solutions when embedded into diverse groups. The panel members and RSE participants shared various practices working well for them:
Utilising a single Teams site with multiple channels to disseminate information about computation and data in research.
Conducting awareness sessions about available technologies and solutions.
Establishing GPU/coding clubs (R, Python, Git).
Organizing showcase events.
Hosting a Festival of Ideas.
Podcasts to share stories.
Hosting a series of clinics with Support Officers to educate PIs on the role of research IT and RSEs in projects and associated costing considerations.

**Reply by mattjbr123** [reply]
As part of the informal [Hydrological Data Science Network](https://teams.microsoft.com/l/team/19%3AhQGUyjV3ORW2dK-HX0RPur3wbbxfbZrEtZbNxzq9oDY1%40thread.tacv2/conversations?groupId=e1e30737-933b-4603-a0ae-9aa653acd27a&tenantId=7f025136-39f3-4f2c-a3c6-cbb1ce21b36a) we used to hold sortof awareness sessions where someone would talk about something new/cool/interesting/useful they've found/done. Am thinking that perhaps this could be a good model to build on

**Reply by mattjbr123** [reply]
P.S. Lmk if you can't access the teams team I might have to add you manually or approve a request or something

**Reply by metazool** [reply]
I was able to join without approval, which is lovely!

**Reply by mattjbr123** [reply]
Excellent!

**Comment by metazool** [comment]
Thank you for rediscovering and bumping this thread, @mattjbr123 

After our collective chat with Fran from People Ops yesterday i started an issue for the [smallest thing we can do to start](https://github.com/NERC-CEH/rse_group/issues/43) - which is to trial a "software clinic" model at our own site. In Edinburgh there's already a Teams-based "Data Users" informal group which runs an occasional show-and-tell event. I know "communities of practise" should help cross the four sites, but "providing a reinforcing reason to visit the office and get serendipitous benefits from doing that" is a meaningful motivator...

I'm hoping to enlist @dcmo and/or others on the data stewardship track in forming a core group who would go to a fortnightly, hour-long, in-person mutual support session and think of topics we could offer as livecode-style introductions to specific topics (packaging, testing, etc)

**Reply by mattjbr123** [reply]
I very much like this idea, and think that aligns well with the Hydrological Data Science Network meetings we used to have. So for Wallingford site I could try restarting those in a "software clinic"ish format and perhaps broaden them out a bit to not just hydrology and/or data science folks, depending on uptake/group size.

**Comment by mattjbr123** [comment]
Continuing the discussion over at https://github.com/orgs/ukceh-rse/discussions/2
